File: 2050-count-good-numbers/count-good-numbers.js
/**
 * @param {number} n
 * @return {number}
 */
var countGoodNumbers = function (n) {
    const mod = BigInt(1e9 + 7);

    function power(base, exp, mod) {
        let ans = 1n;
        base = BigInt(base) % mod;
        exp = BigInt(exp);

        while (exp > 0) {
            if (exp % 2n === 1n) {
                ans = (ans * base) % mod;
            }
            base = (base * base) % mod;
            exp /= 2n;
        }

        return ans;
    }

    let even = Math.floor((n + 1) / 2);
    let odd = Math.floor(n / 2);

    let count = (power(5, even, mod) * power(4, odd, mod)) % mod;

    return Number(count);
};

File: 2050-count-good-numbers/count-good-numbers.cpp
class Solution {
public:
    long long Power(long long base, long long exp, long long mod) {
        long long ans = 1;
        base %= mod;

        while (exp > 0) {
            if (exp % 2 == 1) {
                ans = (ans * base) % mod;
            }
            base = (base * base) % mod;
            exp /= 2;
        }

        return ans;
    }

    int countGoodNumbers(long long n) {
        long long even = 0, odd = 0;
        long long mod = 1e9 + 7;

        if (n % 2 == 0) {
            even = n / 2;
            odd = n / 2;
        } else {
            even = n / 2 + 1;
            odd = n / 2;
        }

        long long ans = (Power(5, even, mod) * Power(4, odd, mod)) % mod;

        return ans;
    }
};